proxy/src/main/java/net/md_5/bungee/util/ChatComponentTransformer.java

```diff
@@ -63,7 +64,7 @@ public static ChatComponentTransformer getInstance()
         {
             if ( root.getExtra() != null && !root.getExtra().isEmpty() )
             {
-                List<BaseComponent> list = Lists.newArrayList( transform( player, root.getExtra().toArray( new BaseComponent[ root.getExtra().size() ] ) ) );
+                List<BaseComponent> list = new ArrayList<>( Arrays.asList( transform( player, root.getExtra().toArray( new BaseComponent[root.getExtra().size()] ) ) ) );
```

Comment:
I suggest to provide a second, private transform method which operatores on `Collection<BaseComponent>` to get rid of the array - list transformations and just call `transform(player, new ArrayList<>(root.getExtra))`

Reply:
Maybe something like this would be better?
```
ListIterator<BaseComponent> iterator = root.getExtra().iterator();
while (iterator.hasNext()) {
  BaseComponent component = iterator.next();
  component = transform(player, component)[0]
  iterator.set(component);
}
```

Reply:
I'd leave this unresolved, I'm going to use the ListIterator for now, if you are sure that it is okay then you resolve that yourself.